src/operator/elemwise_op_common.h

```diff
@@ -63,13 +71,13 @@ inline bool ElemwiseShape(const nnvm::NodeAttrs& attrs,
     attrs, in_attrs, out_attrs, TShape());
 }
 
-template<int n_in, int n_out>
+template<int n_in, int n_out, int total_in = n_in, int total_out = n_out>
```

Comment:
This is hacky. It's better to make a new shape function and call ElemwiseAttr from it.
You are currently not checking the type of weigh32